test/test_raw_api.py

```diff
@@ -0,0 +1,94 @@
+######################################################################
+#
+# File: b2/test_raw_api.py
+#
+# Copyright 2018 Backblaze Inc. All Rights Reserved.
+#
+# License https://www.backblaze.com/using_b2_code.html
+#
+######################################################################
+
+from __future__ import print_function
+
+import six
+
+from b2.raw_api import B2RawApi
+from b2.b2http import B2Http
+from b2.exception import UnusableFileName
+from .test_base import TestBase
+
+if six.PY3:
+    unichr = chr
+
+# Unicode characters for testing filenames.  (0x0394 is a letter Delta.)
+TWO_BYTE_UNICHR = unichr(0x0394)
+CHAR_UNDER_32 = unichr(31)
+DEL_CHAR = unichr(127)
+
+
+class TestRawAPIFilenames(TestBase):
+    """Test that the filename checker passes conforming names and rejects those that don't."""
+
+    def _should_be_ok(self, filename):
+        """Call with test filenames that follow the filename rules.
+
+        :param filename: unicode (or str) that follows the rules"""
+        print(u"Filename \"{}\" should be OK".format(filename))
+        self.assertIsNone(self.raw_api.check_b2_filename(filename))
+
+    def _should_raise(self, filename, exception_message):
+        """Call with filenames that don't follow the rules (so the rule checker should raise).
+
+        :param filename: unicode (or str) that doesn't follow the rules
+        :param exception_message: regexp that matches the exception's detailed message
+        """
+        print(
+            u"Filename \"{}\" should raise UnusableFileName(\".*{}.*\")."
+            .format(filename, exception_message)
+        )
+        with self.assertRaisesRegexp(UnusableFileName, exception_message):
+            self.raw_api.check_b2_filename(filename)
+
+    def test_b2_filename_checker(self):
+        """Test a conforming and non-conforming filename for each rule.
+
+        From the B2 docs (https://www.backblaze.com/b2/docs/files.html):
+        - Names can be pretty much any UTF-8 string up to 1024 bytes long.
+        - No character codes below 32 are allowed.
+        - Backslashes are not allowed.
+        - DEL characters (127) are not allowed.
+        - File names cannot start with "/", end with "/", or contain "//".
+        - Maximum of 250 bytes of UTF-8 in each segment (part between slashes) of a file name.
+        """
+        print("test b2 filename rules")
+        self.raw_api = B2RawApi(B2Http())
```

Comment:
This would normally go in the `setUp()` method in a unit test, which is used for constructor-ish things.  For each test method, the framework makes an instance of the test class, calls `setUp()`, and then calls the test method.  The other option would be to put this in the constructor.

It's nice to set all instance variables in either the constructor or setUp(); in a language without declarations, it helps document what the instance variables are.

The other option here would be to make `check_b2_filename` be a class method, and then not have to create an instance of `B2RawApi`.